Update difficulty thresholds (#2102)
ppy/osu-wiki: wiki/Difficulties/en.md

![Star rating ranges](SR_range.png "Star rating ranges")

> For the [Ranking Criteria](/wiki/Ranking_Criteria), star ratings typically correspond to these difficulties:

Reviewer: Was there a reason to change `From` to `For`?

can probably omit `typically` from the sentence if they are going to be defined that way in https://github.com/ppy/osu-web/pull/4181

Author: "For" because the ranking criteria don't actually define the rating brackets (this page does), and "typically" because it's completely acceptable for difficulties to be out of the listed ranges.

Reviewer: ok

> - ![Easy icon](/wiki/shared/diff/easy-s.png "Easy icon") Easy: 1.0★–2.0★

Reviewer: ```suggestion
- ![Easy icon](/wiki/shared/diff/easy-s.png "Easy icon") Easy: 0.0★–1.99★
```

Author: For the 1.0 to 0.0 change, I'd argue that *very* few maps go below 1.0★ and all typically are named "Beginner" or have special names. The graphs in ppy/osu-web#4181 show fuzziness around the borders.

As for the .99 change, I think the precision is unnecessary and arbitrary, in my opinion—it would be most correct to say <2.0★ etc., but I think just specifying ranges without providing a [strict inequality](https://en.wikipedia.org/wiki/Inequality_(mathematics)) makes reading it a tiny bit more difficult for no additional benefit.

Reviewer 2: the .99 thing can go either way, but you should use 0 for the lower bound here because we're only concerned about the icon in this section, and 0-1 stars does use the Easy icon

Reviewer: > For the 1.0 to 0.0 change, I'd argue that very few maps [...]

I suppose so, but weren't these going to list out the icons that would be used on the site? Sure, they'd be called "Beginner", but the site would still use a green icon.

> As for the .99 change, I think [...]

Probably so, but if going off of this article, would a beatmap with 2.0★ be both Easy and Normal? Again, from above, weren't these going to list out the icons that would be used on the site?

Reviewer 2: yeah that's a good point I'd go with the .99 then. (and all of MAP's related suggestions)

also stays consistent with how the article was previously written, which seemed to serve its purpose fine

Author: Fair enough.

> - ![Normal icon](/wiki/shared/diff/normal-s.png "Normal icon") Normal: 2.0★–2.7★

Reviewer: ```suggestion
- ![Normal icon](/wiki/shared/diff/normal-s.png "Normal icon") Normal: 2.0★–2.69★
```

> - ![Hard icon](/wiki/shared/diff/hard-s.png "Hard icon") Hard: 2.7★–4.0★

Reviewer: ```suggestion
- ![Hard icon](/wiki/shared/diff/hard-s.png "Hard icon") Hard: 2.7★–3.99★
```

> - ![Insane icon](/wiki/shared/diff/insane-s.png "Insane icon") Insane: 4.0★–5.3★

Reviewer: ```suggestion
- ![Insane icon](/wiki/shared/diff/insane-s.png "Insane icon") Insane: 4.0★–5.29★
```

> - ![Expert icon](/wiki/shared/diff/expert-s.png "Expert icon") Expert: 5.3★–6.5★

Reviewer: ```suggestion
- ![Expert icon](/wiki/shared/diff/expert-s.png "Expert icon") Expert: 5.3★–6.49★
```

- ![Expert+ icon](/wiki/shared/diff/expertplus-s.png "Expert+ icon") Expert+: >6.5★

Note that these ranges are not limits; maps may deviate from the specified ranges significantly and still be acceptable.

### Web icons

> The above star rating ranges determines which web icon is going to be used on any given beatmap. Following the icon ranges would usually give you a good estimate to achieve a balanced difficulty spread. There are some guidelines (see [the Mapset section of the Ranking Criteria](/wiki/Ranking_Criteria/#mapset)) you can follow to improve the difficulty spread of your beatmapset.

Reviewer: ```suggestion
The star rating ranges from above determines which web icon is going to be used on any given beatmap. Following those ranges would usually give you a good estimate to achieve a balanced difficulty spread. There are some guidelines (see [the Mapset section of the Ranking Criteria](/wiki/Ranking_Criteria/#mapset)) you can follow to improve the difficulty spread of your beatmapset.
```

maybe?

Reviewer 2: ```suggestion
The above star rating ranges determine which web icon is going to be used on any given beatmap. Following the icon ranges will usually give you a good estimate to achieve a balanced difficulty spread. There are some guidelines (see [the Mapset section of the Ranking Criteria](/wiki/Ranking_Criteria/#mapset)) you can follow to improve the difficulty spread of your beatmapset.
```

> ## Difficulty levels

Reviewer 3: Shouldn't the difficulty level section here be updated as well as the main difficulty naming article? ie add expert+, hell oni, etc.

Author: There are no pages existing for the Expert+ difficulties for any mode, and I don't know if I should create them just to put "anything goes within reasonable bounds" on them.

Reviewer 3: I think it would be more unusual for there to be inconsistencies between the two pages

Reviewer 2: they should be added at some point but it should also be more fleshed out than a simple note. I'll open an issue for it, we can deal with that in a later PR
